qt-qml/test/suite/qml-debug.test.mts

```
@@ -0,0 +1,625 @@
+} from '../helper.mts';
+import {
+  prepareQmlBreakpointsFromMarkers,
+  addBreakpoints,
+  makeQmlDebugConfig,
+  startDebugAndWaitForStop,
+  stopDebugSession,
+  evaluateExpression
+} from '../debug-helper.mts';
+
+const FS_SETTLE_DELAY_MS = 500;
+const DISK_FLUSH_DELAY_MS = 1000;
+
+// ---------------------------------------------------------------------------
+// QML Debugger Integration Tests
+// ---------------------------------------------------------------------------
+// This test suite validates QML debugging functionality including:
+// 1. Breakpoint hits and stack traces
+// 2. Variable evaluation via DAP evaluate request
+// 3. Execution control (continue, step, etc.)
+//
+// Important notes about QML debugging:
+// - QML object properties (counter, message, etc.) are NOT JavaScript local variables
+// - They do NOT appear in DAP scopes (Global, Local, Closure scopes)
+// - QML properties MUST be accessed via DAP 'evaluate' request
+// - JavaScript local variables in QML functions DO appear in scopes
+//
+// Run with: QT_TEST_DEBUG=1 npm run test:qt-qml -- --qt-root="/path/to/Qt"
+// ---------------------------------------------------------------------------
+
+// ---------------------------------------------------------------------------
+// Shared debug logging + sandbox lifecycle for this test file
+// ---------------------------------------------------------------------------
+const DEBUG = process.env.QT_TEST_DEBUG === '1';
+const dlog = (...args: unknown[]) => {
+  if (DEBUG) console.log(...args);
+};
+
+describe('QML Debugger integration', function () {
+  this.timeout(150_000); // QML debugging can take longer
+  let sb: sinon.SinonSandbox;
+
+  setupSandboxLifecycleHooks(
+    (_sb) => (sb = _sb),
+    async () => {
+      dlog('[qml-debug] Activating Qt extensions');
+      await activateQtCore();
+      await activateQtQml();
+    }
+  );
+
+  it('configures, builds, and debugs Qt Quick app with QML debugging', async function () {
```

I notice only the first test performs, cleanBuildDir, writing CMakePresets.json, cmake.setConfigurePreset, cmake.configure, cmake.build, the following tests assume the build directory is present and only call makeQmlDebugConfig() and launch the debugger.

This is ok because Mocha runs tests in order and the first test will fail early if configure/build does not work. But... if downstream test is run in isolation or if the tests are ever reordered this will break.

I strongly recomand to extract the “ensure configured + built” logic into a before('configure and build QML project', ...) that does the full configure/build once.	Then the tests 2–7 can rely on that.

---

I updated tests. Can you take a look at them?

---

thanks